programs/openbook-v2/src/state/orderbook/book.rs

```diff
@@ -267,16 +267,16 @@ impl<'a> Orderbook<'a> {
 
         // Record the taker trade in the account already, even though it will only be
         // realized when the fill event gets executed
-        let mut taker_fees = 0_u64;
+        let mut taker_fees_native = 0_u64;
```

Comment:
why not calculate taker_fees_lots here as well?

Reply:
I think it makes sense to compute the fees in native units to have high resolution on them. Otherwise sub-lot fees would need to be rounded: but can't round up (otherwise openbook would possibly charge 1 lot taker fees on a 1 lot order), and rounding down would unnecessarily reduce collected fees.

We could compute lots and native taker_fees independently, but imo the correctness is clearer to see when the low-resolution value is derived from the high-resolution one.